packages/mock-block-dock/src/mock-block-dock.tsx

```diff
@@ -229,25 +236,13 @@ export const MockBlockDock: FunctionComponent<MockBlockDockProps> = ({
       setEntityIdOfEntityForBlock={setEntityIdOfEntityForBlock}
       updateEntity={graphServiceCallbacks.updateEntity}
     >
-      {!debugMode ? (
-        <Box>
-          {!hideDebugToggle && (
-            <Box className={styles["mbd-debug-mode-toggle-header"]}>
-              <button
-                className={styles["mbd-debug-mode-toggle"]}
-                type="button"
-                onClick={() => setDebugMode(true)}
-              >
-                Preview Mode
-                <OffSwitch />
-              </button>
-            </Box>
-          )}
-          {Component}
-        </Box>
-      ) : (
-        <DebugView>{Component}</DebugView>
-      )}
+      <Suspense>
+        {hideDebugToggle && !debugMode ? (
```

Comment:
Optional suggestion (out of scope): we can rename `hideDebugToggle` to `pure` or `plain` to communicate the intent a bit better.